module_utils/ansible_nailgun_cement.py

```diff
@@ -42,34 +43,50 @@ def create_server(server_url, auth, verify_ssl):
 
 # Common functionality to manipulate entities
 def naildown_entity_state(entity_class, entity_dict, entity, state, module):
+    """ Ensure that a given entity has a certain state """
     changed = False
     if state == 'present':
-        if len(entity) == 0:
+        if entity is None:
             changed = create_entity(entity_class, entity_dict, module)
     elif state == 'latest':
-        if len(entity) == 0:
+        if entity is None:
             changed = create_entity(entity_class, entity_dict, module)
         else:
-            changed = update_entity(entity[0], entity_dict, module)
+            changed = update_entity(entity, entity_dict, module)
     else:
         # state == 'absent'
-        if len(entity) != 0:
-            changed = delete_entity(entity[0], module)
+        if entity is not None:
+            changed = delete_entity(entity, module)
     return changed
 
 
-def find_entity(entity_class, **kwargs):
+def find_entities(entity_class, **kwargs):
+    """ Find entities by certain criteria """
     return entity_class().search(
         query={'search': ','.join(['{0}="{1}"'.format(
-            key, kwargs[key]) for key in kwargs])}
+            key, kwargs[key]) for key in kwargs]), 'per_page': sys.maxint}
```

Comment:
"Fix nailgun" ;)
No, seriously, let's move that discussion to https://github.com/SatelliteQE/nailgun/ and see what they say?